src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -29,14 +36,18 @@ internal class HttpChannelFactory<TChannel>
         private AuthenticationSchemes _authenticationScheme;
         private HttpCookieContainerManager _httpCookieContainerManager;
         private HttpClient _httpClient;
+        volatile MruCache<string, string> _credentialHashCache;
+        private volatile MruCache<string, HttpClient> _httpClientCache;
         private int _maxBufferSize;
         private SecurityCredentialsManager _channelCredentials;
+        SecurityTokenManager _securityTokenManager;
```

Comment:
NIT: access modifiers needed. 

Question only: volatile needed because we're doing the double-locked initialization, correct?